### PR TITLE
Replace references to IRC with Zulip

### PR DESCRIPTION
**Pull request by jeremywiebe**
This PR just updates the various project docs to point to Zulip instead of IRC. 

Resolves (for this repo): https://github.com/xi-editor/xi-editor/issues/961

### CODE REVIEW

**Comment by jeremywiebe** [comment]
@nangtrongvuon Good idea.  I'll get a PR for the main repo as well.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The xi-editor project follows the Rust Code of Conduct, reproduced below.
 **Contact**: [mods@xi-editor.io](mailto:mods@xi-editor.io)
 
 * We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.
-* On IRC, please avoid using overtly sexual nicknames or other nicknames that might detract from a friendly, safe and welcoming environment for all.
+* On Zulip, please avoid using overtly sexual nicknames or other nicknames that might detract from a friendly, safe and welcoming environment for all.
```
Comment (cmyr) [inline CODE_OF_CONDUCT.md]: I wonder if it just makes sense to say something like, "On chat platforms (like Zulip or IRC).."

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -17,9 +17,7 @@ The xi-editor project follows the Rust Code of Conduct, reproduced below.
 
 ## Moderation
 
-
-These are the policies for upholding our community's standards of conduct. If you feel that a thread needs moderation, please contact
-a mod (on IRC) or a maintainer (on Github). To contact a moderator by email, use [mods@xi-editor.io](mailto:mods@xi-editor.io).
+These are the policies for upholding our community's standards of conduct. If you feel that a thread needs moderation, please contact a maintainer on Github. To contact a moderator by email, use [mods@xi-editor.io](mailto:mods@xi-editor.io).
```
Comment (cmyr) [inline CODE_OF_CONDUCT.md]: github doesn't really have 'private messaging' so I'm not sure this makes sense. Maybe just list the email?

Comment (jeremywiebe) [reply CODE_OF_CONDUCT.md]: Yes, that's probably more clear. I was thinking you could look up an email through Github, but that's alot of indirection. I'll fix this.

**Comment by cmyr** [comment]
Looks good, thanks @jeremywiebe!

**Pull request merged**